### ISSUE TITLE
Sticky Warrants

### ISSUE THREAD

**Issue by maackle** [holochain/holochain#3641]
When responding to a `hdk::get` (et al), send the Warrant instead of the data if it was invalid. Same for cascade retrievals.

The recipient will store any Warrant received in the cache which is keyed by the op hash of its target, so that for every new occurrence of data, the warrant store can be quickly checked for the existence of a warrant.

These might be good data types to use in place of existing Option<_> responses:

```rust

enum InternalGetResponse<D> {
    Data(D),  // -> Some(D)
    Warranted(Warrant),  // -> None
    NoData,  // -> None
}

enum RetrieveResponse<D> {
    Data(D, CascadeSource),
    Warranted(Warrant, CascadeSource),
    NoData,
}

```

**Comment by abe-njama** [comment]
Related PR: Refactor Warrant to include author and timestamp https://github.com/holochain/holochain/pull/4020

**Comment by maackle** [comment]
Closed by #4033
